### PR TITLE
Resolve "Add to wallet on Checkout confirmation #10339"

### LINKED ISSUE #10339: Add to wallet on Checkout confirmation
We need to add a button to "add to wallet" on the final screen in the checkout confirmation!
We already have code to add to wallet on the keychain and we could just easily re-use this code here.

### PR DESCRIPTION
**Pull request by SVell**
# Description

- Fixed watchAsset in useProvider to add ERC721 tokens
- Updated watchAsset call in keychain
- Added "Add to crypto wallet button" on checkout page after mint

# Issues

Fixes #10339 
Refs https://github.com/unlock-protocol/unlock/issues/10339

# Checklist:

- [+] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [+] I have commented my code, particularly in hard-to-understand areas
- [+] I have updated the docs to reflect my changes if applicable
- [+] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [+] I have performed a self-review of my own code
- [+] If my code involves visual changes, I am adding applicable screenshots to this thread

## Release Note Draft Snippet

- Fixed watchAsset in useProvider to add ERC721 tokens
- Updated watchAsset call in keychain
- Added "Add to crypto wallet button" on checkout page after mint

AddToWallet

### CODE REVIEW

**Comment by julien51** [comment]
Thanks a lot @SVell ! Let me look at this closely tomorrow!

**Comment by SVell** [comment]
Hey @julien51, any progress on verifying the changes? Since it is my first pull request to the Unlock, do not want to start any other bug bounty until this one is resolved. 
Btw, not really sure why the unit tests have failed. Moreover, double-checked the failure logs and got more confused...

**Comment by julien51** [comment]
CI is failing because of this:
```
./src/hooks/useProvider.ts:12:1
Type error: 'string' is declared but its value is never read.

  10 | import { getCurrentNetwork } from '~/utils/session'
  11 | import { useConnectModal } from './useConnectModal'
> 12 | import { string } from 'zod'
     | ^
  13 | 
  14 | export interface EthereumWindow extends Window {
  15 |   web3: any
1
Error: Process completed with exit code 1.```

**Comment by SVell** [comment]
@julien51 all checks complete, my apologies for the inconvenience

File: unlock-app/src/components/interface/checkout/main/Minting.tsx [hunk 1]
```diff
@@ -103,65 +106,83 @@ export const MintingScreen = ({
...
-              </AddToDeviceWallet>
-            </li>
-          )}
-        </ul>
+        <>
+          <div>
+            <button
+              className="w-full px-2 h-8 text-xs flex justify-between items-center text-center rounded-md bg-black text-white gap-2 font-bold"
+              onClick={() => {
+                console.log(tokenId)
```
Comment (julien51) [inline unlock-app/src/components/interface/checkout/main/Minting.tsx:114]: Cleanup please :)

**Comment by julien51** [comment]
ok the code looks good, but I am a bit worried about clutter on the UI. 

Here are a few ideas on how to simplify: 

First move the link to explorer like that:

image

**Comment by julien51** [comment]
Also I thinl we could combine the 3 "add to wallet" buttons in a single one where the icon is ▼ and the options appear as drowndown. this should simplify the UI a lot.

Let me know what you think!

**Review by julien51 — COMMENTED** [review]
Code looks good!

File: unlock-app/src/components/interface/checkout/main/Minting.tsx (hunk 1 above)
Comment (SVell) [reply unlock-app/src/components/interface/checkout/main/Minting.tsx:114]: Will add a new component tomorrow

**Comment by SVell** [comment]
> Also I thinl we could combine the 3 "add to wallet" buttons in a single one where the icon is ▼ and the options appear as drowndown. this should simplify the UI a lot.
> 
> Let me know what you think!

That really makes sense, will also see what can be done and whether or not it will fit the UI

**Comment by SVell** [comment]
Screenshot 2024-02-09 095556
Screenshot 2024-02-09 095541

**Comment by julien51** [comment]
I'm not sure the button position is good but I like the general idea. Also can you make the button smaller (small or tiny variant).
Thanks

**Comment by ChialiT** [comment]
image


how about this?
place all the buttons in same location, 
I think adding mobile wallets those features are quite nice, we can show them up with 1 clicks only.

**Comment by julien51** [comment]
Ho nice!! cc @SVell ^^ I like that very much!

**Comment by SVell** [comment]
> Ho nice!! cc @SVell ^^ I like that very much!

Ok, on it

**Comment by SVell** [comment]
> image
> 
> 
> 
> 
> 
> how about this?
> 
> place all the buttons in same location, 
> 
> I think adding mobile wallets those features are quite nice, we can show them up with 1 clicks only.

Btw, is this a mobile friendly solution?

**Comment by SVell** [comment]
Screenshot 2024-02-12 111501
Screenshot 2024-02-12 111439

**Review by julien51 — APPROVED** [review]
ok LGTM!

**Pull request merged**